lib/node_modules/@stdlib/array/base/cusome-by-right/docs/types/index.d.ts

```diff
@@ -0,0 +1,151 @@
+*/
+type Unary<T, U> = ( this: U, value: T ) => boolean;
+
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @param value - collection value
+* @param index - collection index
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Binary<T, U> = ( this: U, value: T, index: number ) => boolean;
+
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @param value - collection value
+* @param index - collection index
+* @param collection - input collection
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Ternary<T, U> = ( this: U, value: T, index: number, collection: Collection<T> ) => boolean;
+
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @param value - collection value
+* @param index - collection index
+* @param collection - input collection
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Predicate<T, U> = Nullary<U> | Unary<T, U> | Binary<T, U> | Ternary<T, U>;
+
+/**
+* Interface describing `cusomeByRight`.
+*/
+interface CusomeByRight {
+	/**
+	* Cumulatively tests whether at least n array elements in a provided array pass a test implemented by a predicate function, while iterating from right-to-left.
+	*
+	* @param x - input array
+	* @param n - number of elements
+	* @param predicate - test function
+	* @param thisArg - execution context
+	* @returns output array
+	*
+	* @example
+	* function isPositive( v ) {
+	* 	return v > 0;
+	* }
+	* var x = [ 0, 1, 1, 0, 0 ];
+	*
+	* var y = cusomeByRight( x, 2, isPositive );
+	*  // returns [ false, false, false, true, true ];
+	*/
+	<T = unknown, U = unknown>( x: Collection<T> | AccessorArrayLike<T>, n: number, predicate: Predicate<T, U>, thisArg?: ThisParameterType<Predicate<T, U>> ): Array<boolean>;
+
+	/**
+	* Cumulatively tests whether at least n array elements in a provided array pass a test implemented by a predicate function, while iterating from right-to-left and assigns the results to the provided output array.
+	*
+	* @param x - input array
+	* @param n - number of elements
+	* @param y - output array
+	* @param stride - output array stride
+	* @param offset - output array offset
+	* @param predicate - test function
+	* @param thisArg - execution context
+	* @returns output array
+	*
+	* @example
+	* function isPositive( v ) {
+	* 	return v > 0;
+	* }
+	* var x = [ 1, 1, 0, 0, 0 ];
+	* var y = [ false, null, false, null, false, null, false, null, false, null ];
+	*
+	* var arr = cusomeByRight.assign( x, 2, y, 2, 0, isPositive );,
+	* // returns [ false, null, false, null, false, null, false, null, true, null ];
+	*/
+	assign<T, U extends Collection<T> | AccessorArrayLike<T>>( x: Collection | AccessorArrayLike<unknown>, n: number, y: U, stride: number, offset: number, predicate: Predicate<T, U>, thisArg?: ThisParameterType<Predicate<T, U>> ): Array<boolean>;
```

Comment:
The return type is incorrect. 

cc @Planeshifter

Reply:
I changed return type to `U`